src/scenic/syntax/scenic.gram

```diff
@@ -701,6 +701,37 @@ class_def_raw[ast.ClassDef]:
         )
      }
 
+scenic_class_def_block:
+    | NEWLINE INDENT a=scenic_class_statements DEDENT { a }
+    | simple_stmts
+    | invalid_block
+
+scenic_class_statements[list]: a=scenic_class_statement+ { list(itertools.chain.from_iterable(a)) }
+
+scenic_class_statement[list]:
+    | a=scenic_class_property_stmt { [a] }
+    | a=compound_stmt { [a] }
+    | a=scenic_stmts { a }
+    | a=simple_stmts { a }
+
+scenic_class_property_stmt:
+    # not a simple statement; reads NEWLINE
+    | a=NAME b=['[' attrs=','.scenic_class_property_attribute+ ']' { attrs } ] ':' c=expression NEWLINE { 
+        s.PropertyDef(
+            property=a.string,
+            attributes=b if b is not None else [],
+            value=c,
+            LOCATIONS,
+        )
+     }
+
+# fail if `NAME [ <expr> ]` pattern is found at top level of class definition and
+# <expr> is neither `additive` nor `dynamic`
+scenic_class_property_attribute: &&(
```

Comment:
[`&&`](https://we-like-parsers.github.io/pegen/grammar.html#e-6) is an operator that fails immediately if the expression after that fails to parse.

If what follows after `[` is not `additive` or `dynamic`, it raises a syntax error.
<img width="446" alt="Screen Shot 2022-07-14 at 9 08 17" src="https://user-images.githubusercontent.com/10496155/179027551-42c10bf8-29de-416f-8c63-eddd5aed4581.png">

Reply:
I first wrote a rule like the following:

```
scenic_class_property_attribute:
    | "additive" { s.Additive(LOCATIONS) }
    | "dynamic" { s.Dynamic(LOCATIONS) }
    | !NAME
```

However, if you put a non-Name expression inside the square brackets (like quoted strings or numbers), this rule would fail but will be parsed as a normal annotated assignment.

We can list possible tokens and fail all of them, but I think it would be better to use `&&` here.

Reply:
Agreed.